### ISSUE TITLE
Enabling Key Vault soft-delete is not working on existing resource

### ISSUE THREAD

**Issue by kadirercetin** [hashicorp/terraform-provider-azurerm#16289]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

0.14

### AzureRM Provider Version

v2.75.0

### Affected Resource(s)/Data Source(s)

azurerm_key_vault

### Terraform Configuration Files

```hcl
resource "azurerm_key_vault" "key_vault" {
  name                = "kyvlttestname"
  location            = "northeurope",
  resource_group_name = "rgtest"

  lifecycle {
    ignore_changes = [access_policy]
  }
  enabled_for_deployment = false
  enabled_for_disk_encryption = false
  enabled_for_template_deployment = false

  purge_protection_enabled    = true
  soft_delete_retention_days  = 7
}
```

### Debug Output/Panic Output

```shell
In generated state file, I can see that "soft_delete_enabled": true
```

### Expected Behaviour

Soft delete enabled on Key Vault

### Actual Behaviour

Soft delete is not enabled on existing Key Vault.

State file after terraform apply;
![2022-04-07 15_39_01-C__Users_N16293 NEWDAY_Downloads_fdprr ngd01 tfstate - Notepad++](https://user-images.githubusercontent.com/81424471/162230218-73a3ed69-4cc8-41d6-8e70-e83830942607.png)

![2022-04-07 15_43_27-keyvtngdfdprr0101 - Microsoft Azure](https://user-images.githubusercontent.com/81424471/162226277-4a1a535f-6632-419b-b5dd-082571068107.png)

### Steps to Reproduce

1- Create Azure Key Vault with old provider version ("2.36.0")  

```
provider "azurerm" {
   version = "2.36.0"
}

resource "azurerm_key_vault" "key_vault" {
...
}
```

2- Update azurerm provider and set "purge_protection_enabled" = true, which key vault soft-delete will be enabled as default according to that [documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault#purge_protection_enabled)
```
provider "azurerm" {
   version = "2.75.0"
}

resource "azurerm_key_vault" "example" {
  name                        = "examplekeyvault"
  location                    = azurerm_resource_group.example.location
  resource_group_name         = azurerm_resource_group.example.name
  enabled_for_disk_encryption = true
  tenant_id                   = data.azurerm_client_config.current.tenant_id

  soft_delete_retention_days  = 7
  purge_protection_enabled    = true
}


```

3- terraform plan 
(which is not showing that soft-delete will be enabled)

4- terraform apply

**Comment by sinbai** [comment]
@kadirercetin  thank you for opening this issue. I think this issue is by design. Per the [terraform provider doc v2.36](https://registry.terraform.io/providers/hashicorp/azurerm/2.36.0/docs/resources/key_vault#soft_delete_enabled) ,the default value of  `soft_delete_enabled` property is false. I assume in your use case the KV was created with v2.36 of terraform provider and there is no "soft_delete_enabled = true" in tf config, right? If so, per the [API doc](https://docs.microsoft.com/en-us/rest/api/keyvault/keyvault/vaults/create-or-update#vaultproperties) for the `enablePurgeProtection` property, the `enablePurgeProtection` is effective only if soft delete is also enabled. This means that the `purge_protection_enabled` was not effective when  the "purge_protection_enabled= true" was configurated with v2.75 of terraform provider in your use case. 

For the soft_delete_enabled = true in state file you mentioned, according to the description in the [terraform provider doc v2.75  ](https://registry.terraform.io/providers/hashicorp/azurerm/2.75.0/docs/resources/key_vault) => "_Version v2.42 of the Azure Provider and later ignore the value of the soft_delete_enabled field and force this value to be true_", which means after v2.42, the value of `soft_delete_enabled ` would be set to true in the state file and it is not supported to be updated. 

Based on the above, as a solution, could you try to update `soft_delete_enabled` to true first with versions prior to v2.42 of terraform provider, and then update `purge_protection_enabled` to true with v2.75 of terraform provider?

**Comment by kadirercetin** [comment]
@sinbai thank you for the answer. As in the description of [terraform provider doc v2.75 ](https://registry.terraform.io/providers/hashicorp/azurerm/2.75.0/docs/resources/key_vault)=> _"Version v2.42 of the Azure Provider and later ignore the value of the soft_delete_enabled field and force this value to be true"_, I'm just expecting `soft_delete_enabled` set to `true `regardless of the value state file or with which version of provider KV has been created.

It is not possible for us to downgrade the provider to be able to set this up. Is there any other option?

**Comment by sinbai** [comment]
@kadirercetin thanks for your feedback. I have submitted a PR to set `soft_delete_enabled` to true when updating the KV. So please track of it to see if you would unblocked once it's released.

**Comment by tombuildsstuff** [comment]
Fixed via #16368
